### PR TITLE
Feature/update readme

### PR DESCRIPTION
**Pull request by pbodsk**
I've taken a stab at updating the README for this project, hoping that I've made it more easy to grasp what is needed to setup `Keychain` in a project.

I would like to have more examples on how to actually use the Keychain in the last part so if you have any...don't hold back 😄 

Also...feel free to comment on the format or content....feuer frei 😄

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,45 +114,44 @@ struct UserResetKeychainConfig: KeychainConfig, Equatable {
...
-            }
-    }
+With all the setup out of the way, it is time to kick back and take advantage of `Keychain`. You can now use the `UserAccessKeychainConfig`, `UserRefreshKeychainConfig` and `UserResetKeychainConfig` objects that you created previously to generate JWT tokens by calling the `makeToken(on:, currentDate:)`
 
-    func forgotPassword(request: Request) -> EventLoopFuture<HTTPStatus> {
-        UserForgotPasswordRequest
-            .sendToken(on: request, currentDate: currentDate())
-            .transform(to: .accepted)
-    }
+Here is an example on how to generate a new refreshToken.
```
Comment (cweinberger) [inline README.md]:
```suggestion
Here is an example on how to generate a new `refreshToken`.
```

**Review by cweinberger — APPROVED** [review]
I like this RW tone of voice :D

**Review by heidipuk — APPROVED** [review]
As christian said, the flow of the text is really nice :D

File: README.md [hunk 2]
```diff
@@ -28,30 +27,27 @@ targets: [
...
-)
-app.keychain.configure(
-    signer: JWTSigner(
-        algorithm: TestJWTAlgorithm(name: UserResetKeychainConfig.jwkIdentifier.string)
-    ),
-    config: UserResetKeychainConfig()
-)
-```
+These are the steps required to use Keychain in your project.
+1. Define a Payload which must conform to the `KeychainPayload` protocol
```
Comment (siemensikkema) [inline README.md]:
```suggestion
1. Define a Payload that conforms to the `KeychainPayload` protocol
```

File: README.md [hunk 3]
```diff
@@ -28,30 +27,27 @@ targets: [
...
+2. Create `KeychainConfig` objects for the key types you would like to use
+3. Configure your `Keychain` using a `Signer` and the `KeychainConfig` objects defined in step 2
+4. Actually start using your `Keychain`
+
+Time to look at each step in detail
```
Comment (siemensikkema) [inline README.md]:
```suggestion
Time to look at each step in detail.
```

File: README.md [hunk 4]
```diff
@@ -28,30 +27,27 @@ targets: [
...
+
+### Define a Payload
+
+Your payload must conform to the  `KeychainPayload` protocol, meaning that it must contain:
+- `init(expirationDate: Date, user: User) throws`
+- `func findUser(request: Request) -> EventLoopFuture<User>` which is where you do a search for the user you were presented in the `init` method
+- `func verify(using signer: JWTSigner) throws` which will verify that your token is still valid
+
+Furthermore you need to tell your `KeychainPayload` what its `associatedtype` for User translates to.
```
Comment (siemensikkema) [inline README.md]:
```suggestion
Furthermore you need to tell your `KeychainPayload` what its `associatedtype` for `User` translates to.
```

File: README.md [hunk 5]
```diff
@@ -28,30 +27,27 @@ targets: [
...
+
+Here is an example that uses elements from a JWT token and verifies that the `ext` part is not expired. Note that `findUser` in this case only returns a test user. In real life you probably want to do a lookup somewhere where users are stored.
```
Comment (siemensikkema) [inline README.md]:
```suggestion
Here is an example that uses elements from a JWT token and verifies that the expiration (`exp`) claim is not expired. Note that `findUser` in this case only returns a test user. In real life you probably want to do a lookup somewhere where users are stored.
```

File: README.md [hunk 6]
```diff
@@ -71,11 +67,24 @@ struct UserJWTPayload: KeychainPayload {
...
+        try exp.verifyNotExpired()
     }
 }
 ```
-#### KeychainConfigs
+
+### Create `KeychainConfig` Objects
+
+Your `KeychainConfig` objects must contain:
+- an identifier (eg: access, refresh or reset)
```
Comment (siemensikkema) [inline README.md]:
```suggestion
- an identifier (eg: access, refresh or reset): `jwkIdentifier`
```

File: README.md [hunk 7]
```diff
@@ -71,11 +67,24 @@ struct UserJWTPayload: KeychainPayload {
...
+- an expirationTimeInterval
```
Comment (siemensikkema) [inline README.md]:
```suggestion
- an `expirationTimeInterval`
```

File: README.md [hunk 8]
```diff
@@ -71,11 +67,24 @@ struct UserJWTPayload: KeychainPayload {
...
+
+And you need to connect your `KeychainConfig` with the `KeychainPayload` you defined in step 1 (the `KeychainConfig` has a `typealias` for a `KeychainPayload`).
+
+Here is an example creating three `KeychainConfig` objects:
+- A `UserAccessKeychainConfig` with the identifier "access" and a expirationTimeInterval of 300 seconds
+- A `UserRefreshKeychainConfig` with the identifier "refresh" and a expirationTimeInterval of 600 seconds
+- A `UserResetKeychainConfig` with the identifier "reset" and a expirationTimeInterval of 400 seconds
```
Comment (siemensikkema) [inline README.md]:
```suggestion
- A `UserAccessKeychainConfig` with the identifier "access" and an `expirationTimeInterval` of 300 seconds
- A `UserRefreshKeychainConfig` with the identifier "refresh" and an `expirationTimeInterval` of 600 seconds
- A `UserResetKeychainConfig` with the identifier "reset" and an `expirationTimeInterval` of 400 seconds
```

File: README.md [hunk 9]
```diff
@@ -105,45 +114,44 @@ struct UserResetKeychainConfig: KeychainConfig, Equatable {
...
+    signer: JWTSigner(
+        algorithm: TestJWTAlgorithm(name: UserResetKeychainConfig.jwkIdentifier.string)
+    ),
+    config: UserResetKeychainConfig()
+)
+```
 
-struct UserController {
-    let currentDate: () -> Date
+Note the `signer` parameter. You can use one of the built in signers as in the first example where we use the `.hs256` signer with a key. or you can provide your own signer as it is done in the last two examples.
```
Comment (siemensikkema) [inline README.md]:
```suggestion
Note the `signer` parameter. You can use one of the built-in signers as in the first example where we use the `.hs256` signer with a key. Alternatively, you can provide your own signer as it is done in the last two examples.
```

**Review by siemensikkema — COMMENTED** [review]
Really nice! Just a few nits but otherwise let's get this in! (finally)

**Review by siemensikkema — APPROVED** [review]

**Pull request merged**